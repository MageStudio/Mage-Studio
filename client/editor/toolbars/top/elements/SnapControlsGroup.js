import React from 'react';
import classnames from 'classnames';
import { Input, Switch } from 'antd';
import { MinusOutlined, PlusOutlined } from '@ant-design/icons';

import style from '../toolbar.module.scss';

const SnapControlsGroup = ({ enabled, snap, onSnapIncrease, onSnapDecrease, onSnapEnabledChange }) => {
    const snapSettingsContainerClassname = classnames(
        style['controls-settings-group'],
        style['snap-settings-group'], {
            [style.enabled]: enabled
        });
    const snapSettingsInputClassname = classnames(style['snap-settings-group-item'], style['snap-settings-group-item-text-input']);

    return (
        <div className={snapSettingsContainerClassname}>
            <label className={style['snap-settings-group-item']}>Snap:</label>
            <Switch
                onChange={onSnapEnabledChange}
                className={style['snap-settings-group-item']}
                size="small"
                checked={enabled} />
            <Input
                size="small"
                placeholder="1"
                value={snap}
                className={snapSettingsInputClassname}
                prefix={<MinusOutlined onClick={onSnapDecrease} />}
                suffix={<PlusOutlined onClick={onSnapIncrease} />}/>
        </div>
    );
};

export default SnapControlsGroup;