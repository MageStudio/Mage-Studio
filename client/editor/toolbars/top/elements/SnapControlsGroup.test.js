import { iteratee } from 'lodash';
import React from 'react';
import sinon from 'sinon';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import SnapControlsGroup from './SnapControlsGroup';

describe('SnapControlsGroup', () => {

    it('renders properly', () => {
        const component = shallow(<SnapControlsGroup />);
        expect(toJSON(component)).toMatchSnapshot();
    });

    describe('behaviours', () => {

        it.skip('onSnapIncrease', () => {
            const spy = sinon.spy();
            const component = shallow(<SnapControlsGroup onSnapIncrease={spy} />);

            component.find('PlusOutlined').simulate('click');

            expect(spy.called).toBe(true);
        });

        it.skip('onSnapDecrease', () => {
            const spy = sinon.spy();
            const component = shallow(<SnapControlsGroup onSnapDecrease={spy} />);

            component.find('MinusOutlined').simulate('click');

            expect(spy.called).toBe(true);
        });

        it('onSnapEnabledChange', () => {
            const spy = sinon.spy();
            const component = shallow(<SnapControlsGroup onSnapEnabledChange={spy} />);

            component.find('Switch').simulate('change');

            expect(spy.called).toBe(true);
        });
    });
});
