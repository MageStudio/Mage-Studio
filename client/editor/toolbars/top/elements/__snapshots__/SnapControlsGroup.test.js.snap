// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapControlsGroup renders properly 1`] = `
<div
  className="controls-settings-group snap-settings-group"
>
  <label
    className="snap-settings-group-item"
  >
    Snap:
  </label>
  <Switch
    className="snap-settings-group-item"
    size="small"
  />
  <Input
    className="snap-settings-group-item snap-settings-group-item-text-input"
    placeholder="1"
    prefix={<ForwardRef(MinusOutlined) />}
    size="small"
    suffix={<ForwardRef(PlusOutlined) />}
    type="text"
  />
</div>
`;
